Create upgrade-monitoring-services.md (#14523)
pingcap/docs: upgrade-monitoring-services.md

---
title: Upgrade Cluster Monitoring Services
summary: Learn how to upgrade the Prometheus, Grafana, and Alertmanager monitoring services for your TiDB cluster.
---

# Upgrade TiDB Cluster Monitoring Services

When you deploy a TiDB cluster using TiUP, TiUP automatically deploys monitoring services (such as Prometheus, Grafana, and Alertmanager) for the cluster. If you scale out this cluster, TiUP also automatically adds monitoring configurations for newly added nodes during the scaling.

> Pay attention that TiUP uses its own configurations to override the configurations of the monitoring services. If you directly upgrade the monitoring services by replacing their configuration files, any subsequent operations such as `deploy`, `scale-out`, `scale-in`, or `reload` on the cluster might overwrite your upgrade, leading to errors. To upgrade Prometheus, Grafana, and Alertmanager, you can follow the upgrade steps in this document.

Reviewer: ```suggestion
Pay attention as TiUP uses its own configurations to override the configurations of the monitoring services. If you directly upgrade the monitoring services by replacing their configuration files, any subsequent operations such as `deploy`, `scale-out`, `scale-in`, or `reload` on the cluster might overwrite your upgrade, leading to errors. To upgrade Prometheus, Grafana, and Alertmanager, you can follow the upgrade steps in this document.
```

> **Note:**
>

> > - If your monitoring services are not deployed and managed by TiUP, you can directly upgrade them without referring to this document.

Reviewer: Can we add a link to the guide that describes how to configure a TiDB cluster to use an external Grafana/Prometheus setup?

Reviewer 2: Do you refer to this one?
https://docs.pingcap.com/tidb/stable/deploy-monitoring-services

Reviewer: Yes

Author: added in https://github.com/pingcap/docs/pull/14523/commits/2f3c904cf80016662dd4a620821caed987fdf20c

> > - The TiDB compatibility with newer versions of monitoring services has not been not tested, so it is possible that some features might not work as expected after the upgrade. For any issues, you can create an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Reviewer: So if people upgrade they don't have a supported version anymore?

Reviewer 2: The official supported version is the one in the tidb package. Monitoring is provided by 3rd party, we do not have resources to test the different versions . We will do some fix if users found any high impact issues.

Reviewer: The problem I see with this is with vulnerabilities that are fixed upstream but not in the version that is in the tidb package.  Then the user could upgrade, but then they'll lose support.

```

After the upgrade, you can go to the home page of the Prometheus server (usually at `http://<Prometheus-server-host-name>:9090`), click **Status** in the top navigation menu, and then open the **Runtime & Build Information** page to check the Prometheus version and confirm whether the upgrade is successful.

## Upgrade Grafana

For better compatibility with TiDB, it is recommended to use the Grafana installation package provided in the TiDB installation package. The version of Grafana in the TiDB installation package is fixed. If you want to use a newer Grafana version, refer to [Grafana Release Notes](https://grafana.com/docs/grafana/latest/whatsnew/) for new features of each version and choose a suitable version for your production environment. You can also consult with PingCAP technical staff for a recommended version.

In the following upgrade steps, you need to download the Grafana installation package of your desired version from the Grafana website, and then use it to create a Grafana package that TiUP can use.

### Step 1. Download a new Grafana installation package from the Grafana website

> Download a new installation package from the [Grafana download page](https://grafana.com/grafana/download?pg=get&plcmt=selfmanaged-box1-cta1) and extract it.

Reviewer: This website lists two versions: "OSS" (opensource) and "Enterprise". I think we should suggest people to use the "OSS" edition.

Reviewer 2: I don't think we  should add this suggestion here. Customers can choose anyone they want.

Reviewer: If both should work fine, then we can add a note about that: e.g. "Both the OSS and Enterprise versions work, choose the one that matches your setup".

Author: updated in https://github.com/pingcap/docs/pull/14523/commits/2f3c904cf80016662dd4a620821caed987fdf20c

    ```bash
    tar -xzf grafana-v{version}-linux-amd64.tar.gz
    ```

### Step 3. Create a new Grafana package that TiUP can use

1. Copy the files extracted in [Step 1](#step-1-download-a-new-grafana-installation-package-from-the-grafana-website), and then use the copied files to replace the files in the `./grafana-v{version}-linux-amd64/` directory extracted in [Step 2](#step-2-download-the-grafana-installation-package-provided-by-tidb).
2. Recompress the `./grafana-v{version}-linux-amd64` directory and name the new compressed package as `grafana-v{new-version}.tar.gz`, where `{new-version}` can be specified according to your need.

    ```bash
    cd grafana-v{version}-linux-amd64.tar.gz
    tar -zcvf ../grafana-v{new-version}.tar.gz ./
    ```

### Step 4. Upgrade Grafana using the newly created Grafana package

Execute the following command to upgrade Grafana:

```bash
tiup cluster patch <cluster-name> grafana-v{new-version}.tar.gz -R grafana

```

After the upgrade, you can go to the home page of the Grafana server (usually at `http://<Grafana-server-host-name>:3000`), and then check the Grafana version on the page to confirm whether the upgrade is successful.

> ## Upgrade AlertManager

Reviewer: Can we add instructions for how to check the alertmanager version after the upgrade?

Author: added in https://github.com/pingcap/docs/pull/14523/commits/fa5d462feb511afe08df8d65c146f1133459042d

Reviewer: Perfect. Thanks.

The AlertManager package in the TiDB installation package is directly from the Prometheus website. Therefore, when upgrading AlertManager, you only need to download and install a new version of AlertManager from the Prometheus website.

### Step 1. Download a new AlertManager installation package from the Prometheus website

Download the `alertmanager` installation package from the [Prometheus download page](https://prometheus.io/download/#alertmanager).

### Step 2. Upgrade AlertManager using the downloaded installation package

Execute the following command to upgrade AlertManager:

```bash
tiup cluster patch <cluster-name> alertmanager-v{new-version}-linux-amd64.tar.gz -R alertmanager

> ```

Reviewer 2: ```suggestion
```

After the upgrade, you can go to the home page of the AlertManager server (usually at `http://<Alertmanager-server-host-name>:9093`), click **Status** in the top navigation menu, and then check the AlertManager version and confirm whether the upgrade is successful.
```

1. Download the `TiDB-community-server` package from the [TiDB download page](https://pingcap.com/product/#SelectProduct) and extract it.
2. In the extracted files, locate `prometheus-v{version}-linux-amd64.tar.gz` and extract it.

    ```bash
    tar -xzf prometheus-v{version}-linux-amd64.tar.gz
    ```

### Step 3. Create a new Prometheus package that TiUP can use

1. Copy the files extracted in [Step 1](#step-1-download-a-new-prometheus-installation-package-from-the-prometheus-website), and then use the copied files to replace the files in the `./prometheus-v{version}-linux-amd64/prometheus` directory extracted in [Step 2](#step-2-download-the-prometheus-installation-package-provided-by-tidb).
2. Recompress the `./prometheus-v{version}-linux-amd64` directory and name the new compressed package as `prometheus-v{new-version}.tar.gz`, where `{new-version}` can be specified according to your need.

    ```bash
    cd prometheus-v{version}-linux-amd64.tar.gz
    tar -zcvf ../prometheus-v{new-version}.tar.gz ./
    ```

### Step 4. Upgrade Prometheus using the newly created Prometheus package

Execute the following command to upgrade Prometheus:

```bash

> tiup cluster patch <cluster-name> prometheus-{new-version}.tar.gz -R prometheus

Author: ```suggestion
tiup cluster patch <cluster-name> prometheus-v{new-version}.tar.gz -R prometheus
```

```

After the upgrade, you can go to the home page of the Prometheus server (usually at `http://<Prometheus-server-host-name>:9090`), click **Status** in the top navigation menu, and then open the **Runtime & Build Information** page to check the Prometheus version and confirm whether the upgrade is successful.

## Upgrade Grafana

For better compatibility with TiDB, it is recommended to use the Grafana installation package provided in the TiDB installation package. The version of Grafana in the TiDB installation package is fixed. If you want to use a newer Grafana version, refer to [Grafana Release Notes](https://grafana.com/docs/grafana/latest/whatsnew/) for new features of each version and choose a suitable version for your production environment. You can also consult with PingCAP technical staff for a recommended version.

In the following upgrade steps, you need to download the Grafana installation package of your desired version from the Grafana website, and then use it to create a Grafana package that TiUP can use.

### Step 1. Download a new Grafana installation package from the Grafana website

> Download a new installation package from the [Grafana download page](https://grafana.com/grafana/download?pg=get&plcmt=selfmanaged-box1-cta1) and extract it.

Author: How about the following updates? @dveeden

```suggestion
1. Download a new installation package from the [Grafana download page](https://grafana.com/grafana/download?pg=get&plcmt=selfmanaged-box1-cta1). You can choose either the `OSS` or `Enterprise` edition according to your needs.
2. Extract the downloaded package. 
```

> **Note:**
>
> - If your monitoring services are [deployed manually](deploy-monitoring-services.md) instead of using TiUP, you can directly upgrade them without referring to this document.
> - The TiDB compatibility with newer versions of monitoring services has not been not tested, so it is possible that some features might not work as expected after the upgrade. For any issues, you can create an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
> - The upgrade steps in this document are applicable for TiUP version 1.9.0 and later. Therefore, check your TiUP version before the upgrade.
> - When you use TiUP to upgrade the TiDB cluster, TiUP will redeploy the monitoring services to the default version. You need to redo the upgrade for monitoring services after the TiDB upgrade. 

> ## Upgrade Prometheus

Author: ```suggestion

## Upgrade Prometheus
```

---
title: Upgrade Cluster Monitoring Services
summary: Learn how to upgrade the Prometheus, Grafana, and Alertmanager monitoring services for your TiDB cluster.
---

# Upgrade TiDB Cluster Monitoring Services

When deploying a TiDB cluster, TiUP automatically deploys monitoring services (such as Prometheus, Grafana, and Alertmanager) for the cluster. If you scale out this cluster, TiUP also automatically adds monitoring configurations for newly added nodes during the scaling. The monitoring services automatically deployed by TiUP are usually not the latest versions of these third-party monitoring services. To use the latest versions, you can follow this document to upgrade the monitoring services.

> When managing a cluster, TiUP uses its own configurations to override the configurations of the monitoring services. If you directly upgrade the monitoring services by replacing their configuration files, any subsequent TiUP operations such as `deploy`, `scale-out`, `scale-in`, or `reload` on the cluster might overwrite your upgrade, leading to errors. To upgrade Prometheus, Grafana, and Alertmanager, follow the steps in this document rather than directly replacing configuration files.

Reviewer 3: ```suggestion
When managing a cluster, TiUP uses its own configurations to override the configurations of the monitoring services. If you directly upgrade the monitoring services by replacing their configuration files, any subsequent TiUP operations such as `deploy`, `scale-out`, `scale-in`, and `reload` on the cluster might overwrite your upgrade, leading to errors. To upgrade Prometheus, Grafana, and Alertmanager, follow the steps in this document rather than directly replacing configuration files.
```

> **Note:**
>

> > - If your monitoring services are [deployed manually](deploy-monitoring-services.md) instead of using TiUP, you can directly upgrade them without referring to this document.

Author: ```suggestion
> - If your monitoring services are [deployed manually](/deploy-monitoring-services.md) instead of using TiUP, you can directly upgrade them without referring to this document.
```

> > - The TiDB compatibility with newer versions of monitoring services has not been not tested, so it is possible that some features might not work as expected after the upgrade. For any issues, you can create an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Reviewer 3: ```suggestion
> - The TiDB compatibility with newer versions of monitoring services has not been tested, so it is possible that some features might not work as expected after the upgrade. For any issues, create an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

## Upgrade Grafana

For better compatibility with TiDB, it is recommended to use the Grafana installation package provided in the TiDB installation package. The version of Grafana in the TiDB installation package is fixed. If you want to use a newer Grafana version, refer to [Grafana Release Notes](https://grafana.com/docs/grafana/latest/whatsnew/) for new features of each version and choose a suitable version for your production environment. You can also consult with PingCAP technical staff for a recommended version.

In the following upgrade steps, you need to download the Grafana installation package of your desired version from the Grafana website, and then use it to create a Grafana package that TiUP can use.

### Step 1. Download a new Grafana installation package from the Grafana website

1. Download a new installation package from the [Grafana download page](https://grafana.com/grafana/download?pg=get&plcmt=selfmanaged-box1-cta1). You can choose either the `OSS` or `Enterprise` edition according to your needs.
2. Extract the downloaded package. 

### Step 2. Download the Grafana installation package provided by TiDB

> 1. Download the `TiDB-community-server` package from the [TiDB download page](https://www.pingcap.com/download) and extract it.

Author: ```suggestion
1. Download the TiDB **Server Package** package from the [TiDB download page](https://www.pingcap.com/download) and extract it.
```

> 1. Download the `TiDB-community-server` package from the [TiDB download page](https://pingcap.com/product/#SelectProduct) and extract it.

Reviewer 3: ```suggestion
1. Download the `TiDB-community-server` package from the [TiDB download page](https://www.pingcap.com/tidb/) and extract it.
```

    ```bash
    tar -xzf grafana-v{version}-linux-amd64.tar.gz
    ```

### Step 3. Create a new Grafana package that TiUP can use

1. Copy the files extracted in [Step 1](#step-1-download-a-new-grafana-installation-package-from-the-grafana-website), and then use the copied files to replace the files in the `./grafana-v{version}-linux-amd64/` directory extracted in [Step 2](#step-2-download-the-grafana-installation-package-provided-by-tidb).
2. Recompress the `./grafana-v{version}-linux-amd64` directory and name the new compressed package as `grafana-v{new-version}.tar.gz`, where `{new-version}` can be specified according to your need.

    ```bash
    cd grafana-v{version}-linux-amd64.tar.gz
    tar -zcvf ../grafana-v{new-version}.tar.gz ./
    ```

### Step 4. Upgrade Grafana using the newly created Grafana package

Execute the following command to upgrade Grafana:

```bash
tiup cluster patch <cluster-name> grafana-v{new-version}.tar.gz -R grafana

```

After the upgrade, you can go to the home page of the Grafana server (usually at `http://<Grafana-server-host-name>:3000`), and then check the Grafana version on the page to confirm whether the upgrade is successful.

> ## Upgrade AlertManager

Reviewer 3: ```suggestion
## Upgrade Alertmanager
```

> ## Upgrade AlertManager

Reviewer 3: ```suggestion
## Upgrade Alertmanager
```

> The AlertManager package in the TiDB installation package is directly from the Prometheus website. Therefore, when upgrading AlertManager, you only need to download and install a new version of AlertManager from the Prometheus website.

Reviewer 3: ```suggestion
The Alertmanager package in the TiDB installation package is directly from the Prometheus website. Therefore, when upgrading Alertmanager, you only need to download and install a new version of Alertmanager from the Prometheus website.
```

> The AlertManager package in the TiDB installation package is directly from the Prometheus website. Therefore, when upgrading AlertManager, you only need to download and install a new version of AlertManager from the Prometheus website.

Reviewer 3: ```suggestion
The Alertmanager package in the TiDB installation package is directly from the Prometheus website. Therefore, when upgrading Alertmanager, you only need to download and install a new version of Alertmanager from the Prometheus website.
```

> ### Step 1. Download a new AlertManager installation package from the Prometheus website

Reviewer 3: ```suggestion
### Step 1. Download a new Alertmanager installation package from the Prometheus website
```

> ### Step 1. Download a new AlertManager installation package from the Prometheus website

Reviewer 3: ```suggestion
### Step 1. Download a new Alertmanager installation package from the Prometheus website
```

Download the `alertmanager` installation package from the [Prometheus download page](https://prometheus.io/download/#alertmanager).

> ### Step 2. Upgrade AlertManager using the downloaded installation package

Reviewer 3: ```suggestion
### Step 2. Upgrade Alertmanager using the downloaded installation package
```

> ### Step 2. Upgrade AlertManager using the downloaded installation package

Reviewer 3: ```suggestion
### Step 2. Upgrade Alertmanager using the downloaded installation package
```

> Execute the following command to upgrade AlertManager:

Reviewer 3: ```suggestion
Execute the following command to upgrade Alertmanager:
```

> Execute the following command to upgrade AlertManager:

Reviewer 3: ```suggestion
Execute the following command to upgrade Alertmanager:
```

```bash
tiup cluster patch <cluster-name> alertmanager-v{new-version}-linux-amd64.tar.gz -R alertmanager
```

> After the upgrade, you can go to the home page of the AlertManager server (usually at `http://<Alertmanager-server-host-name>:9093`), click **Status** in the top navigation menu, and then check the AlertManager version to confirm whether the upgrade is successful.

Reviewer 3: ```suggestion
After the upgrade, you can go to the home page of the Alertmanager server (usually at `http://<Alertmanager-server-host-name>:9093`), click **Status** in the top navigation menu, and then check the Alertmanager version to confirm whether the upgrade is successful.
```

> After the upgrade, you can go to the home page of the AlertManager server (usually at `http://<Alertmanager-server-host-name>:9093`), click **Status** in the top navigation menu, and then check the AlertManager version to confirm whether the upgrade is successful.

Reviewer 3: ```suggestion
After the upgrade, you can go to the home page of the Alertmanager server (usually at `http://<Alertmanager-server-host-name>:9093`), click **Status** in the top navigation menu, and then check the Alertmanager version to confirm whether the upgrade is successful.
```
